File: app/src/main/java/com/google/android/stardroid/sensors/SensorModelAdaptor.java
package com.google.android.stardroid.sensors;

import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;

import com.google.android.stardroid.control.AstronomerModel;

import javax.inject.Inject;

/**
 * Connects the rotation vector to the model code.
 */
public class SensorModelAdaptor implements SensorEventListener {
  private AstronomerModel model;

  @Inject
  SensorModelAdaptor(AstronomerModel model) {
    this.model = model;
  }

  @Override
  public void onSensorChanged(SensorEvent event) {
    // do something with the model
  }

  @Override
  public void onAccuracyChanged(Sensor sensor, int accuracy) {
    // Do nothing.
  }
}

File: app/src/main/java/com/google/android/stardroid/sensors/SensorModelAdaptor.kt
package com.google.android.stardroid.sensors

import android.hardware.Sensor
import android.hardware.SensorEvent
import android.hardware.SensorEventListener
import com.google.android.stardroid.control.AstronomerModel
import javax.inject.Inject

/**
 * Connects the rotation vector to the model code.
 */
class SensorModelAdaptor @Inject internal constructor(private val model: AstronomerModel) :
  SensorEventListener {
  override fun onSensorChanged(event: SensorEvent) {
    // do something with the model
  }

  override fun onAccuracyChanged(sensor: Sensor, accuracy: Int) {
    // Do nothing.
  }
}